src/server/pfs/s3/server.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gogo/protobuf/types"
+
+	"github.com/pachyderm/pachyderm/src/client"
+)
+
+const locationResponse = `<?xml version="1.0" encoding="UTF-8"?>
+<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/">PACHYDERM</LocationConstraint>`
+
+type handler struct {
+	pc *client.APIClient
+}
+
+func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	parts := strings.SplitN(r.URL.Path, "/", 4)
+
+	if len(parts) == 2 && parts[1] == "_ping" {
+		// matches `/_ping`
+		w.Write([]byte("OK"))
+	} else if len(parts) == 3 && parts[2] == "" {
+		// matches `/repo/`
+		h.serveRoot(w, r, parts[1])
+	} else if len(parts) == 4 {
+		// matches /repo/branch/path/to/file.txt
+		repo := parts[1]
+		branch := parts[2]
+		file := parts[3]
+		h.serveFile(w, r, repo, branch, file)
+	} else {
+		http.Error(w, "not found", 404)
```

Comment:
I generally prefer to use `http.StatusNotFound` in place of hardcoded values. Although it's a minor preference since at this point `404` has kind of transcended magic number status and is probably easier to read than `http.StatusNotFound` for most programmers. But as soon as a start adding in some of the more esoteric error codes I want to use the values in the `http` package, and then it just throws off my feng-shui to have some of them be numbers and some constant variables.

Reply:
I agree. I have a separate branch that adds support for writing, and in that I also change hardcoded numbers to `http` constants. Is it OK to defer the change to when I open a PR for that branch?

Reply:
Sure thing.